packages/preview-middleware-client/src/cpe/changes/generic-change.ts

```diff
@@ -131,21 +132,20 @@ interface GenericChangeHandlerReturnType {
     changeTitle: string;
     controlId?: string | string[];
     changeType?: string;
-    configPath?: string;
-    properties: { label: string; value: PropertyValue | Record<string, unknown>; displayValueWithIcon?: boolean }[];
-}
-
-interface GenericChangeHandler {
-    [key: string]: (
-        change: GenericChange,
-        options: ChangeHandlerOptions
-    ) => Promise<GenericChangeHandlerReturnType> | GenericChangeHandlerReturnType;
+    subtitle?: string;
+    properties: { label: string; value?: PropertyValue; displayValueWithIcon?: boolean }[];
 }
+export type ChangeHandler<Change> = (
+    change: Change,
+    options: ChangeHandlerOptions
+) => Promise<GenericChangeHandlerReturnType> | GenericChangeHandlerReturnType;
+type GenericChangeHandler = {
+    [Change in GenericChange as Change['changeType']]: ChangeHandler<Change>;
+};
 
-//: GenericChangeMap
 export const GENERIC_CHANGE_HANDLER: GenericChangeHandler = {
     [ADD_NEW_ANNOTATION_FILE_CHANGE]: (change, { textBundle }) => {
-        const annotationFileChange = change as NewAnnotationFileChange;
+        const annotationFileChange = change;
```

Comment:
Why do we reassign parameters to a variable here? That seems redundant as we can just use a different parameter name.

Reply:
removed